src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+        submitNoSamplesPane_1.add(selectedProblemsLabel);
+        
+        selectProblemsButton = new JButton("Select");
+        selectProblemsButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                showProblemList();
+            }
+        });
+        selectProblemsButton.setToolTipText("Select Problems");
+        submitNoSamplesPane_1.add(selectProblemsButton);
+        
+        JPanel submitLanguages = new JPanel();
+        FlowLayout fl_submitLanguages = (FlowLayout) submitLanguages.getLayout();
+        fl_submitLanguages.setAlignment(FlowLayout.LEFT);
+        submitLanguages.setBorder(new TitledBorder(null, "", TitledBorder.LEADING, TitledBorder.TOP, null, null));
+        submitLanguages.setBounds(33, 191, 650, 32);
+        controlsPane.add(submitLanguages);
+        
+        checkBoxLanguage = new JCheckBox("Languages");
+        checkBoxLanguage.setVerticalAlignment(SwingConstants.TOP);
+        checkBoxLanguage.setToolTipText("Select Languages");
+        checkBoxLanguage.setSelected(true);
+        submitLanguages.add(checkBoxLanguage);
+        
+        submitAllLanguagesRadioButton = new JRadioButton("Submit All Languages");
+        submitAllLanguagesRadioButton.setSelected(true);
+        submitLanguages.add(submitAllLanguagesRadioButton);
+        
+        submitSelectedLanguagesRadioButton = new JRadioButton("Submit Selected");
+        submitLanguages.add(submitSelectedLanguagesRadioButton);
+        
+        selectedLanguagesLabel = new JLabel("None selected Lang?");
+        selectedLanguagesLabel.setToolTipText("None selected");
+        submitLanguages.add(selectedLanguagesLabel);
+        
+        selectLanguageButton = new JButton("Select");
+        selectLanguageButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                showLanguageList();
+            }
+        });
+        selectLanguageButton.setToolTipText("Select Problems");
+        submitLanguages.add(selectLanguageButton);
+        
+
+        languagesButtonGroup.add(submitAllLanguagesRadioButton);
+        languagesButtonGroup.add(submitSelectedLanguagesRadioButton);
+        
+        problemsButtonGroup.add(submitAllProblemsRadioButton);
+        problemsButtonGroup.add(submitSelectedProblemsRadioButton);
+        
+		VersionInfo info = new VersionInfo();
+		String verstring = info.getBuildNumber() + " x " + info.getPC2Version();
+		addLineToTextArea(verstring);
+		
+		
+		
+	}
+
+	protected void showLanguageList() {
+
+	    // TODO show language selection
+	    
+        Language[] Languages = getContest().getLanguages();
+        
+        if (Languages.length == 0) {
+            showMessage("No Languagess defined in contest");
+            return;
+        }
+    
+        JListFrame selectLanguagessFrame = new JListFrame("Select Languages ", Languages, selectedLanguagesIndexes, new ISelectedListsSetter() {
```

Comment:
Ok, picky point -- but others have to read/maintain the code so not completely ridiculous:  `selectLanguagessFrame` was probably intended to be (and probably should be) `selectLanguagesFrame`.